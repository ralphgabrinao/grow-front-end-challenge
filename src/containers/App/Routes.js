import React from 'react';
import { Router } from 'react-router-dom';
import history from '../../history';

const Routes = globalProps => {
	return (
		<Router history={history}>
			<div>
				{/* <AppliedRoute exact path='/' component={Home} props={globalProps} /> */}
			</div>
		</Router>
	);
};

export default Routes;