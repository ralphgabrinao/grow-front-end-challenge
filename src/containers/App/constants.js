export const actionTypes = {
	REDIRECT: 'APP/REDIRECT'
};