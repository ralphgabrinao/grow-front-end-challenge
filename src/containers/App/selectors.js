import { createSelector } from 'reselect';

/* istanbul ignore next */
const getAppState = state => state.App;

export const makeSelectApp = createSelector(
	getAppState,
	appState => { return appState; }
);