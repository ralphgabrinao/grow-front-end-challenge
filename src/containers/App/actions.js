import { actionTypes } from './constants';

const actions = {
	redirect: location => {
		return {
			type: actionTypes.REDIRECT,
			location
		};
	}
};

export default actions;