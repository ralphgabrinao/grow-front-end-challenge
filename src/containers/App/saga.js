import { call, takeEvery } from 'redux-saga/effects';
import { actionTypes } from './constants';
import history from '../../history';

export function * redirect(action) {
	yield call(history.push, action.location);
}

export const appSagas = [
	takeEvery(actionTypes.REDIRECT, redirect)
];