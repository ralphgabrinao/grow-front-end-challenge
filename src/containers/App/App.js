import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { makeSelectApp } from './selectors';
import Routes from './Routes';
import styled from 'styled-components';

const Wrapper = styled.div`
	overflow-y: scroll;
	height: 100vh;
	font-family: "proxima-nova", "Helvetica Neue", Helvetica, Arial, sans-serif;
	background-color: #ECECEC;
	color: #333333;
	line-height: 1.42857143;
	font-size: 12px;
`;

export class App extends React.Component {
	render() {
		const globalProps = {};
		return (
			<Wrapper>
				<Routes globalProps={globalProps}/>
			</Wrapper>
		);
	}
}

App.propTypes = {
	App: PropTypes.any
};

/* istanbul ignore next */
const mapStateToProps = state => {
	return {
		App: makeSelectApp(state)
	};
};


/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
	dispatch
});

export default connect(mapStateToProps, mapDispatchToProps)(App);