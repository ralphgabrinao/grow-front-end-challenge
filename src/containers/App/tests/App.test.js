import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { App } from '../App';
import Routes from '../Routes';

describe('<App />', () => {
	let props;

	beforeEach(() => {
		props = {
			App: {},
			redirect: sinon.spy()
		};
	});

	it('should match snapshot', () => {
		const wrapper = shallow(<App {...props} />);
		expect(wrapper).toBeDefined();
		expect(wrapper).toMatchSnapshot();
	});
});