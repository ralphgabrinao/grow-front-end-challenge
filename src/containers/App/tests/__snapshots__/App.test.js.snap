// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    App={Object {}}
    redirect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "Hello, World!",
        <Routes
          globalProps={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Hello, World!",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "globalProps": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
	height: 100vh;
	font-family: \\"proxima-nova\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;
	background-color: #f7f4ed;
	color: #333333;
	line-height: 1.42857143;
	font-size: 14px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "Hello, World!",
          <Routes
            globalProps={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Hello, World!",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "globalProps": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
	height: 100vh;
	font-family: \\"proxima-nova\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif;
	background-color: #f7f4ed;
	color: #333333;
	line-height: 1.42857143;
	font-size: 14px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
